### DISCUSSION TITLE
Event to skip saving a revision

### DISCUSSION THREAD

**Discussion by ryssbowh** [craftcms/cms#16260] [Ideas] [answered]
Would be useful to allow developers to not save a revision based on an event.

My use case is the following : I have one section that users are allowed to save on the frontend, there's one field in there that I need admins to see the history of, all the rest we don't care about.
I have set the `maxRevisions` setting to 0 to be able to see all history, which creates loads of revisions for this section (and all the others, but that's a different issue : https://github.com/craftcms/cms/discussions/11477), which for most of them aren't needed and are clogging up the database.

With such an event I can calculate field changes myself and skip the revision if needed.

**Comment by brandonkelly** [comment, marked as answer]
Just made this possible for Craft 5.6: (0aa4936df319b6ac837c96e929834942813670de)

```php
use craft\elements\Entry;
use craft\events\RevisionEvent;
use craft\services\Revisions;
use yii\base\Event;

Event::on(
    Revisions::class,
    Revisions::EVENT_BEFORE_CREATE_REVISION,
    function(RevisionEvent $event) {
        if (
            $event->canonical instanceof Entry &&
            $event->canonical->section?->handle === 'myHandle' &&
            Craft::$app->request->isSiteRequest
        ) {
            // avoid creating a revision
            $event->handled = true;
        }
    }
);
```

Note that setting `$event->handled = true` will only prevent a revision from getting created if at least one revision already exists for the element. Otherwise it will be ignored.

**Reply by ryssbowh** [reply]
That was quick ! Thank you

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out now 🚀
